File: src/main/kotlin/studio/pinkcloud/voyager/deployment/cloudflare/responses/DeleteDnsRecord.kt
package studio.pinkcloud.voyager.deployment.cloudflare.responses

import kotlinx.serialization.Serializable

@Serializable
data class DeleteDnsRecordData(
    val id: String,
)

@Serializable
data class DeleteDnsRecordSuccess(
    val result: DeleteDnsRecordData,
)

@Serializable
data class DeleteDnsRecordFailure(
    val errors: List<CloudflareError>,
    val messages: List<CloudflareMessage>,
    val success: Boolean,
)

File: src/modules/cloudflare/types/delete_dns_record.rs
use serde::Deserialize;
use crate::modules::cloudflare::types::cloudflare_responses::CloudflareError;

#[derive(Debug, Deserialize)]
pub struct OkResponse {
  // pub errors: Vec<CloudflareError>,
  // pub messages: Vec<CloudflareMessage>,
  pub result: Option<Data>,
  // pub success: bool,
}

#[derive(Debug, Deserialize)]
pub struct Data {
  pub id: String,
}

#[derive(Debug, Deserialize)]
pub struct FailureResponse {
  pub errors: Vec<CloudflareError>,
  // pub messages: Vec<CloudflareMessage>,
  // pub success: bool,
}
